### PR TITLE
SE-2865: Create a sample notebook for Access Metadata based entitlements

### PR DESCRIPTION
**Pull request by df1116**
# Pull Request Checklist

- [ ] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

This PR adds a notebook to demonstrate the use of Access Metadata based entitlements. It follows the style of the `Property value entitlements` notebook - [here](https://github.com/finbourne/sample-notebooks/blob/master/examples/entitlements/Property%20value%20entitlements.ipynb). There is an existing KB article on using AMD for access control - [here](https://support.lusid.com/knowledgebase/article/KA-01940/en-us). Specifically this notebook makes use of Access Metadata to deny access to externally managed portfolios.

### CODE REVIEW

File: examples/entitlements/Access Metadata entitlements.ipynb [hunk 1]
```diff
@@ -0,0 +1,1190 @@
+{
```
Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Could you add a summary of a use case for doing this here? It is some variation of the text you use in section 3 but with more of a business-user framing. E.g. Want an admin user to be able to see all details but you only want a portfolio manager to see the portfolios that they manage themselves. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #61.    )
</pre><p>You are only checking the status of one of the api factories here - should probably be for all of them if you are being complete. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p><s>The amounts and considerations here are randomised, but typically they would be loaded in externally (for example, from a .csv file).</s></p><p> </p><blockquote>The amounts and considerations for the generated transactions are randomised but typically transactions can come from a number of sources if not generated within LUSID, for example from a .csv file or a 3rd party system.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>I think this explanation needs more detail. It is correct but I think the conciseness means it will be glanced over and it is an important thing to understand. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Similarly to how we set up the notebook at the start, I think this section needs to start with a more business-user framing of the challenge before getting into the technical differences between the two users. It can just be a quick line. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>I would rephrase this slightly because it gives the impression that you are going straight into creating a role using default policies which is not correct; you are actually going to generate the second policy first. </p><p> </p><p>Maybe start by re-iterating the relationship between policies and roles and say that you want to create a role with these two policies (which you are doing) and that the second one does not exist by default and needs to be created first (which you mention but in a less obvious way)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>This is a bit unclear - can you provide a bit more explanation?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Can you add a second line to this which essentially states 'this will give all users the ability to see all portfolios in the scope'. It is just slightly less technical language even though it says the same thing. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>This might read and flow better if this line is more of a heading - just because assigning the role is a multi-stage activity and you don't actually do it until a few cells down where you repeat the text slightly. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Maybe prefix this text with <code>NB:</code> to make sure that the part about the comment is picked up by the reader. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Can you add a second line which restates this in slightly less technical language - e.g. this will restrict access to the portfolio for certain users</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Can this output be formatted a bit nicer?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <blockquote>To entitle User B</blockquote><p>-&gt; To apply the desired restrictions to User B</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (jyaffe) [inline examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Please can this be formatted nicer (same as above)?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (df1116) [reply examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Added a sort of example that the rest of the notebook would be based on.</p>

Comment (df1116) [reply examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Unfortunately, the identity and access <code>ApplicationMetadataApi</code> don't have the <code>get_versions</code> endpoint so it can't be done.</p>

Comment (df1116) [reply examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>Have replaced</p><p> </p><p> </p>

Comment (df1116) [reply examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>I've linked it to the introduction section but not sure it wants more, let me know.</p>

Comment (jyaffe) [reply examples/entitlements/Access Metadata entitlements.ipynb:1]: <p>OK</p>

**Review by jyaffe — APPROVED** [review]

**Comment by jyaffe** [comment]
\LGTM
